### PR TITLE
Fix Travis osx build

### PR DESCRIPTION
**Pull request by ppolewicz**
stacktrace from failed build was:

```
$ if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then virtualenv -p python3 "$HOME/.virtualenv/python3-osx"; fi
Running virtualenv with interpreter /usr/local/bin/python3
Using base prefix '/usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5'
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/virtualenv.py", line 475, in copyfile
    os.symlink(srcpath, dest)
FileExistsError: [Errno 17] File exists: '/usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/config-3.5m' -> '/Users/travis/.virtualenv/python3-osx/lib/python3.5/config-3.5m'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/virtualenv.py", line 2338, in <module>
    main()
  File "/Library/Python/2.7/site-packages/virtualenv.py", line 824, in main
    symlink=options.symlink)
  File "/Library/Python/2.7/site-packages/virtualenv.py", line 984, in create_environment
    site_packages=site_packages, clear=clear, symlink=symlink))
  File "/Library/Python/2.7/site-packages/virtualenv.py", line 1178, in install_python
    copyfile(join(stdlib_dir, fn), join(lib_dir, fn), symlink)
  File "/Library/Python/2.7/site-packages/virtualenv.py", line 478, in copyfile
    copyfileordir(src, dest, symlink)
  File "/Library/Python/2.7/site-packages/virtualenv.py", line 453, in copyfileordir
    shutil.copytree(src, dest, symlink)
  File "/usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/shutil.py", line 309, in copytree
    os.makedirs(dst)
  File "/usr/local/Cellar/python3/3.5.2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/os.py", line 241, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/Users/travis/.virtualenv/python3-osx/lib/python3.5/config-3.5m'
The command "if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then virtualenv -p python3 "$HOME/.virtualenv/python3-osx"; fi" failed and exited with 1 during .
Your build has been stopped.
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ before_install:
   # install python3 on OS X
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install python3; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then virtualenv -p python3 "$HOME/.virtualenv/python3-osx"; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" -a ! -e "$HOME/.virtualenv/python3-osx" ]]; then virtualenv -p python3 "$HOME/.virtualenv/python3-osx"; fi
```
Comment (andrzejgorski) [inline .travis.yml]: `if [[ "$TRAVIS_OS_NAME" == "osx" &&  ! -a "$HOME/.virtualenv/python3-osx" ]]; then virtualenv -p python3 "$HOME/.virtualenv/python3-osx"; fi` missing `&&` operator.

Comment (ppolewicz) [inline .travis.yml]: or a single `[` instead of a double. Fixed.

File: .travis.yml [hunk 2]
```diff
@@ -14,62 +14,146 @@ python:
...
   - if [ -n "$SCA" ]; then pyflakes . || exit; else echo skip; fi
-  - if [ -n "$SCA" ]; then grep -r '\_\_metaclass\_\_' whylog_vim/ && exit 1 || true; else echo skip; fi
-  - if [ -n "$SCA" ]; then exit; else echo skip; fi
-  - python setup.py nosetests
+
+  # isort doesn't get along with yapf. Hopefully we'll figure it out and re-enable in the future.
+  # - if [ -n "$SCA" ]; then isort --order-by-type --recursive --line-width 100 --diff --verbose --check-only || exit; else echo skip; fi
+
+  # home-made bad code detectors
+  - if [ -n "$SCA" ]; then grep -r --extended-regex '^ *class [^\(]+(\(\))?:' whylog/ && exit 1 || true; else echo skip; fi  # old-style class detector
```
Comment (andrzejgorski) [inline .travis.yml]: Change `whylog/` to `whylog_vim`.  It doesn't work now.

File: .travis.yml [hunk 3]
```diff
@@ -14,62 +14,146 @@ python:
...
+  - if [ -n "$SCA" ]; then grep -r 'isinstance\(' . && exit 1 || true; else echo skip; fi  # very non-pythonic
```
Comment (andrzejgorski) [inline .travis.yml:157]: Grep should should takes `whylog_vim` instead of `.`.

File: .travis.yml [hunk 4]
```diff
@@ -14,62 +14,146 @@ python:
...
+  - if [ -n "$SCA" ]; then grep -r --extended-regex '\b(all|any)\(\[' whylog/ && exit 1 || true; else echo skip; fi  # a list is unnecesarily allocated where a generator expression could have been used
```
Comment (andrzejgorski) [inline .travis.yml]: Same as on the line 156.

File: .travis.yml (hunk 3 above)
Comment (ppolewicz) [inline .travis.yml:157]: Why shouldn't it take `.`? This way it is generic and usable in the same form in all repositories.

**Pull request merged**